plugins/jobs/src/js/pages/JobDetailPageContainer.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { componentFromStream, graphqlObservable } from "data-service";
+import { Observable, BehaviorSubject } from "rxjs";
+import gql from "graphql-tag";
+import { default as schema } from "#PLUGINS/jobs/src/js/data/JobModel";
+import Loader from "#SRC/js/components/Loader";
+import Page from "#SRC/js/components/Page";
+import RequestErrorMsg from "#SRC/js/components/RequestErrorMsg";
+import JobDetailPage from "./JobDetailPage";
+
+export const DIALOGS = {
+  EDIT: "edit",
+  DESTROY: "destroy"
+};
+
+const LoadingScreen = function() {
+  return (
+    <Page>
+      <Page.Header breadcrumbs={[]} />
+      <Loader />
+    </Page>
+  );
+};
+
+const ErrorScreen = function() {
+  return (
+    <Page>
+      <Page.Header breadcrumbs={[]} />
+      <RequestErrorMsg />
+    </Page>
+  );
+};
+
+const disabledDialog$ = new BehaviorSubject(null);
+const disableDialog = () => {
+  disabledDialog$.next(DIALOGS.DESTROY);
+};
+
+const jobActionDialog$ = new BehaviorSubject(null);
+
+const handleDestroyButtonClick = () => {
+  jobActionDialog$.next(DIALOGS.DESTROY);
+};
+
+const handleEditButtonClick = () => {
+  jobActionDialog$.next(DIALOGS.EDIT);
+};
+
+const closeDialog = () => {
+  jobActionDialog$.next(null);
+  disabledDialog$.next(null);
+};
+
+const getGraphQL = id =>
+  graphqlObservable(
+    gql`
+      query {
+        job(id: $id) {
+          id
+          name
+          path
+          command
+          schedules
+          cpus
+          description
+          mem
+          docker
+          disk
+          labels
+          jobRuns
+          json
+        }
+      }
+    `,
+    schema,
+    { id }
+  );
+
+export default componentFromStream(props$ => {
+  const id$ = props$.map(props => props.params.id).distinctUntilChanged();
+  const job$ = id$
+    .switchMap(id => getGraphQL(id))
+    .map(({ data: { job } }) => job);
+
+  return Observable.combineLatest([
+    job$,
+    props$,
+    jobActionDialog$,
+    disabledDialog$
+  ])
+    .map(([job, props, jobActionDialog, disabledDialog]) => (
+      <JobDetailPage
+        {...props}
+        job={job}
+        jobActionDialog={jobActionDialog}
+        disabledDialog={disabledDialog}
+        errorMsg={null}
+        handleRunNowButtonClick={() => {}}
```

Comment:
I really don't think it is a good idea we dropped the menu here, and all the callbacks. I can't test the dialog anymore (which is a responsibility of this component). 

I really think this should work with the store in a further ticket, and not just wait for the menu to get done and be integrated later. It also break a bunch of contracts that I am still relying on (which is to some of those handlers that needed to be passed on).

https://github.com/dcos/dcos-ui/pull/3023

I would be ok approving that one, but I would love to see a ticket that integrates and makes it work with 3023, before all the rest of the menu is done.

Reply:
For me, that's what https://jira.mesosphere.com/browse/DCOS-38483 is about.
I limited the scope to no menu items except the edit form, because the dialog was not working previously and I wanted to tackle this explicitly. I am not expecting the actions to work without any more things to be done on this component.